packages/valory/skills/twitter_scoring_abci/rounds.py

```diff
@@ -89,6 +89,26 @@ def latest_mention_tweet_id(self) -> dict:
         return cast(dict, self.db.get_strict("latest_mention_tweet_id"))
 
 
+class OpenAICallCheckRound(CollectSameUntilThresholdRound):
+    """OpenAICallCheckRound"""
+
+    payload_class = TwitterCollectionPayload
+    synchronized_data_class = SynchronizedData
+    none_event = Event.API_ERROR
+
+    CALLS_REMAINING = "CALLS_REMAINING"
+
+    def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
+        """Process the end of the block."""
+        if self.threshold_reached:
+            return self.synchronized_data, Event.DONE
+        if not self.is_majority_possible(
+            self.collection, self.synchronized_data.nb_participants
+        ):
+            return self.synchronized_data, Event.NO_MAJORITY
+        return None
```

Comment:
Still not needed :sweat_smile: 

Take a look at the base class. You also have done and no majority event attributes available.